src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,185 @@
+        int tripHour = Integer.parseInt(tripTimeSplit[0]);
+        int tripMinutes = Integer.parseInt(tripTimeSplit[1]);
+        ZonedDateTime queryZonedDateTime = ZonedDateTime.of(queryDate, LocalTime.of(tripHour, tripMinutes), zoneId);
+
+
+        // Check the dates in a 7-day window starting from the query date.
+        for (int i = 0; i < 7; i++) {
+            ZonedDateTime probedDate = queryZonedDateTime.plusDays(i);
+            if (trip.isActiveOnDate(probedDate)) {
+                result.add(DateTimeUtils.getStringFromDate(probedDate.toLocalDate(), YYYY_MM_DD));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Gets OTP queries to check non-realtime itinerary existence for the given trip.
+     */
+    public static Map<String, String> getItineraryExistenceQueries(MonitoredTrip trip) throws URISyntaxException {
+        return getQueriesFromDates(
+            excludeRealtime(trip.queryParams),
+            getDatesToCheckItineraryExistence(trip)
+        );
+    }
+
+    /**
+     * @return a copy of the specified query, with ignoreRealtimeUpdates set to true.
+     */
+    public static String excludeRealtime(String queryParams) throws URISyntaxException {
+        Map<String, String> params = getQueryParams(queryParams);
+        params.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
+
+        // Insert '?' so others can parse the resulting query string.
+        return "?" + toQueryString(params);
+    }
+
+    /**
+     * Replaces the trip itinerary field value with one from the verified itineraries queried from OTP.
+     * The ui_activeItinerary query parameter is used to determine which itinerary to use,
+     * TODO/FIXME: need a trip resemblance check to supplement the ui_activeItinerary param used in this function.
+     */
+    public static void updateTripWithVerifiedItinerary(MonitoredTrip trip, List<Itinerary> verifiedItineraries) throws URISyntaxException {
+        Map<String, String> params = getQueryParams(trip.queryParams);
+        Itinerary itinerary = null;
+        String itineraryIndexParam = params.get("ui_activeItinerary");
+        if (itineraryIndexParam != null) {
+            try {
+                int itineraryIndex = Integer.parseInt(itineraryIndexParam);
+                itinerary = verifiedItineraries.get(itineraryIndex);
+            } catch (NumberFormatException e) {
+                // TODO: error message.
+            }
+        } else {
+            // FIXME: implement itinerary resemblance to find one matching trip.itinerary.
+        }
+
+        if (itinerary != null) {
+            trip.itinerary = itinerary;
+            trip.initializeFromItinerary();
+        }
+    }
+
+    /**
+     * @return true if the itinerary's startTime is one the same day as the day of the specified date and time.
+     */
+    public static boolean itineraryDepartsSameDay(Itinerary itinerary, String date, String time, ZoneId zoneId) {
```

Comment:
All of this method is oriented around an assumption that the request date and time is relative to a "depart at" kind of query. There needs to be logic related to an "arrive by" query too.

Reply:
I added the arrive-by tests in https://github.com/ibi-group/otp-middleware/pull/77/commits/0aba7da8c43c1e2b26da517ee70d6f821b9a933b.